docs/sources/flow/setup/deploy-agent.md

```diff
@@ -0,0 +1,125 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/setup/deploy-agent/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/setup/deploy-agent/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/setup/deploy-agent/
+canonical: https://grafana.com/docs/agent/latest/flow/setup/start-agent/
+description: Learn about possible deployment topologies for Grafana Agent
+menuTitle: Deploy Grafana Agent
+title: Grafana Agent deployment topologies
+weight: 900
+---
+
+# Deploying Grafana Agent
+
+Grafana Agent is a flexible, vendor-neutral telemetry collector. This
+flexibility means that the Agent doesn’t enforce a specific deployment topology
+but can work in multiple scenarios.
+
+This page lists common topologies used for deployments of Grafana Agent, when
+to consider using each, issues that users may run into, as well as scaling
+considerations.
+
+## As a centralized collection service
+Deploying the Grafana Agent as a centralized service is recommended for
+collecting application telemetry. This allows a smaller set of agents to
+coordinate service discovery, collection, and remote writing.
+
+![](../../../assets/deploy-agent/centralized-collection.png)
+
+Using this topology requires deploying the Agent on separate infrastructure,
+and making sure that agents can discover and reach these applications over the
+network. The main predictor for the size of the agent is the number of active
+metrics series it is scraping; a rule of thumb is ~10kb of memory for each
+series. We recommend you start looking towards horizontal scaling around the 1M
+active series mark.
+
+### Using Kubernetes StatefulSets
+Deploying Grafana Agent as a StatefulSet is the recommended option for metrics
+collection.
+The persistent pod identifiers make it possible to consistently match volumes
+with pods so that you can use them for the WAL directory.
+
+You can also use a Kubernetes deployment in cases where persistent storage is not required, such as a traces-only pipeline.
+
+### Pros
+* Straightforward scaling using [clustering][] or [hashmod sharding][]
+* Minimizes the “noisy neighbor” effect
+* Easy to meta-monitor
+
+### Cons
+* Requires running on separate infrastructure
```

Comment:
Nit: it's not a strict requirement I feel - you can colocate agents on the same workflows. I think that we allow for minimising the noisy-neighbour effect when agents are deployed on separate infrastructure.